openvdb/openvdb/points/AttributeArray.h

```diff
@@ -344,6 +344,11 @@ class OPENVDB_API AttributeArray
     bool operator==(const AttributeArray& other) const;
     bool operator!=(const AttributeArray& other) const { return !this->operator==(other); }
 
+#if OPENVDB_ABI_VERSION_NUMBER >= 9
+    /// Indirect virtual function to retrieve the data buffer cast to a char byte array
+    const char* constDataAsByteArray() const { return this->dataAsByteArray(); }
```

Comment:
Any good reason why this isn't also a virtual function?

Reply:
Well it's redundant no? `Typed` array will have access to both and has a function which returns a `const StorageType*` i.e:

```
const TypedAttributeArray<>* a = ...
const char* cbuf = a->constDataAsByteArray();
auto buf = a->constData();
```

Reply:
I just meant if you wanted to do this:
```
const AttributeArray* a = ...
const char* cbuf = a->constDataAsByteArray();
```
This doesn't really apply for the `constData()` method as you'd need to the know the storage type. However for the `constDataAsByteArray()` method, if we're going to expose it, we might as well expose it on the AttributeArray as a virtual method?